HACKING.md

```diff
@@ -0,0 +1,54 @@
+# How to hack scenario_simulator_v2
+
+## Note
+
+This document is written for those who want to participate in the development of `scenario_simulator_v2` or make their own modifications to `scenario_simulator_v2`. Development requires general knowledge of ROS 2 software and robotic systems and a certain level of C++ skills, as well as reading the ASAM OpenSCENARIO standard documentation appropriately.
+
+## Prerequisite Knowledge
+
+The `scenario_simulator_v2` consists of the following four main components.
+
+### `scenario_test_runner`
+
+The `scenario_test_runner` is responsible for managing the progress of the entire process. It takes a single scenario file or a workflow file of multiple scenarios as arguments, and passes each scenario to `openscenario_interpreter` to evaluate each scenario. Another task of the `scenario_test_runner` is to preprocess each scenario file before running the scenario.
+
+TIER IV uses scenarios in YAML format with some extensions to ASAM OpenSCENARIO 1.2. The extended syntax mainly focuses on parameterization features that were not standardized as of OpenSCENARIO 1.0. Processing of the extended syntax by TIER IV is handled by `scenario_test_runner` and the preprocessed scenarios are XML files that conform to the ASAM OpenSCENARIO 1.2 schema. Note that the scenario passed to `openscenario_interpreter` is therefore in ASAM standard format (`.xosc`), not the extended format by TIER IV (`.yaml`).
+
+Both scenario files written in the TIER IV extended format (`.yaml`) and standard ASAM OpenSCENARIO 1.2 scenario files (`.xosc`) can be given to `scenario_test_runner`.
+
+### `openscenario_interpreter`
+
+ The `openscenario_interpreter` is an interpreter for OpenSCENARIO 1.2 scenario definition files in `.xosc` format, corresponding to the OpenSCENARIO Director in The ASAM OpenSCENARIO architecture. It is a naive syntax tree interpreter that parses a scenario definition file by recursive descent parsing, constructs an abstract syntax tree, and evaluates the syntax tree directly. The `openscenario_interpreter` is primarily responsible for interpreting ASAM OpenSCENARIO, since the fundamental part of the simulation, such as coordinates and time management, is managed by the `traffic_simulator`.
```

Comment:
```suggestion
The `openscenario_interpreter` is an interpreter for OpenSCENARIO 1.2 scenario
definition files in `.xosc` format, corresponding to the OpenSCENARIO Director
in The ASAM OpenSCENARIO architecture. It is a naive syntax tree interpreter
that parses a scenario definition file by recursive descent parsing, constructs
an abstract syntax tree, and evaluates the syntax tree directly. The
`openscenario_interpreter` is primarily responsible for interpreting ASAM
OpenSCENARIO, since the fundamental part of the simulation, such as coordinates
and time management, is managed by the `traffic_simulator`.
```